src/app/rundown-view/services/tv2-piece-group.service.spec.ts

```diff
@@ -5,6 +5,84 @@ import { Tv2Piece } from '../../core/models/tv2-piece'
 import { TestEntityFactory } from '../../test/factories/test-entity.factory'
 
 describe(Tv2PieceGroupService.name, () => {
+  describe(Tv2PieceGroupService.prototype.mergePiecesByStartOffset.name, () => {
+    it('does not produce piece when no piece given', () => {
+      const pieces: Tv2Piece[] = []
+
+      const testee: Tv2PieceGroupService = createTestee()
+      const result: Tv2Piece[] = testee.mergePiecesByStartOffset(pieces)
+
+      const expected: Tv2Piece[] = []
+
+      expect(result).toEqual(expected)
+    })
+
+    it('does not alter piece given a single piece', () => {
+      const testEntityFactory: TestEntityFactory = new TestEntityFactory()
+
+      const pieceOne: Tv2Piece = testEntityFactory.createPiece({
+        name: 'pieceName',
+        start: 123,
+      }) as Tv2Piece
+      const pieces: Tv2Piece[] = [pieceOne]
+
+      const testee: Tv2PieceGroupService = createTestee()
+      const result: Tv2Piece[] = testee.mergePiecesByStartOffset(pieces)
+
+      const expectedPiece: Tv2Piece = testEntityFactory.createPiece({
+        name: 'pieceName',
+        start: 123,
+      }) as Tv2Piece
+      const expected: Tv2Piece[] = [expectedPiece]
+
+      expect(result).toEqual(expected)
+    })
+
+    it('returns single merged piece from two pieces with same start', () => {
+      const testEntityFactory: TestEntityFactory = new TestEntityFactory()
+
+      const pieceOne: Tv2Piece = testEntityFactory.createPiece({
+        name: 'pieceOneName',
+        start: 123,
+      }) as Tv2Piece
+      const pieceTwo: Tv2Piece = testEntityFactory.createPiece({
+        name: 'pieceTwoName',
+        start: 123,
+      }) as Tv2Piece
+      const pieces: Tv2Piece[] = [pieceOne, pieceTwo]
```

Comment:
This might just be a matter of taste, but personally I think that the local variables are unnecessary in this case. Since the creation setup for the pieces are only a couple of lines, it makes it easier to grasp the linearity of the construction – I don't have to go back and forth between the `pieces` list of object references and the instantiation of the references. I would inline the piece creation inside the `pieces` instantiation like so:
```typescript
      const pieces: Tv2Piece[] = [
        testEntityFactory.createPiece({
          name: 'pieceOneName',
          start: 123,
        }),
        testEntityFactory.createPiece({
          name: 'pieceTwoName',
          start: 123,
        }),
      ] as Tv2Piece []
```

I would do the same for the `expected` result.